habitat/sims/habitat_simulator.py

```diff
@@ -392,51 +392,90 @@ def _get_agent_config(self, agent_id: Optional[int] = None) -> Any:
         assert agent_id == 0, "No support of multi agent in {} yet.".format(
             self.__class__.__name__
         )
         return self._sim.get_agent(agent_id).get_state()
 
     def set_agent_state(
         self,
-        position: List[float] = None,
-        rotation: List[float] = None,
+        position: List[float],
+        rotation: List[float],
         agent_id: int = 0,
         reset_sensors: bool = True,
-    ) -> None:
+    ) -> bool:
         """Sets agent state similar to initialize_agent, but without agents
-        creation.
+        creation. On failure to place the agent in the proper position, it is
+        moved back to its previous pose.
 
         Args:
-            position: numpy ndarray containing 3 entries for (x, y, z).
-            rotation: numpy ndarray with 4 entries for (x, y, z, w) elements
-            of unit quaternion (versor) representing agent 3D orientation,
-            (https://en.wikipedia.org/wiki/Versor)
+            position: list containing 3 entries for (x, y, z).
+            rotation: list with 4 entries for (x, y, z, w) elements of unit
+                quaternion (versor) representing agent 3D orientation,
+                (https://en.wikipedia.org/wiki/Versor)
             agent_id: int identification of agent from multiagent setup.
             reset_sensors: bool for if sensor changes (e.g. tilt) should be
                 reset).
+
+        Returns:
+            True if the set was successful else moves the agent back to its
+            original pose and returns false.
         """
         agent = self._sim.get_agent(agent_id)
-        state = self.get_agent_state(agent_id)
-        state.position = position
-        state.rotation = rotation
+        original_state = self.get_agent_state(agent_id)
+        new_state = self.get_agent_state(agent_id)
+        new_state.position = position
+        new_state.rotation = rotation
 
         # NB: The agent state also contains the sensor states in _absolute_
         # coordinates. In order to set the agent's body to a specific
         # location and have the sensors follow, we must not provide any
         # state for the sensors. This will cause them to follow the agent's
         # body
-        state.sensor_states = dict()
+        new_state.sensor_states = dict()
+
+        agent.set_state(new_state, reset_sensors)
 
-        agent.set_state(state, reset_sensors)
+        if not self._check_agent_position(position, agent_id):
+            agent.set_state(original_state, reset_sensors)
+            return False
+        return True
+
+    def get_observations_at(
+        self,
+        position: List[float],
+        rotation: List[float],
+        keep_agent_at_new_pose: bool = False,
+    ) -> Optional[Observations]:
 
-        self._check_agent_position(position, agent_id)
+        # TODO do we need to support multiple agents for this?
+        agent_id = 0
```

Comment:
`agent_id = 0` can be removed, we have default value for that, just `self.get_agent_state()`

Reply:
Done.